### PR TITLE
TASK: mark "dirty" nodes in the trees

### PR DESCRIPTION
**Pull request by dimaip**
Resolves: #806
image

### CODE REVIEW

File: packages/neos-ui/src/Containers/LeftSideBar/NodeTree/Node/index.js [hunk 1]
```diff
@@ -303,6 +307,7 @@ export const ContentTreeNode = withNodeTypeRegistry(connect(
             currentDocumentNode: selectors.UI.ContentCanvas.documentNodeSelector(state),
             focusedNodeContextPath: $get('cr.nodes.focused.contextPath', state),
             toggledNodeContextPaths: selectors.UI.ContentTree.getToggled(state),
+            isNodeDirty: selectors.CR.Workspaces.makeIsContentNodeDirtySelector($get('contextPath', node))(state),
```
Comment (skurfuerst) [inline packages/neos-ui/src/Containers/LeftSideBar/NodeTree/Node/index.js]: same here

File: packages/neos-ui/src/Containers/LeftSideBar/NodeTree/Node/index.js [hunk 2]
```diff
@@ -273,6 +275,7 @@ export const PageTreeNode = withNodeTypeRegistry(connect(
             intermediateContextPaths: selectors.UI.PageTree.getIntermediate(state),
             loadingNodeContextPaths: selectors.UI.PageTree.getLoading(state),
             errorNodeContextPaths: selectors.UI.PageTree.getErrors(state),
+            isNodeDirty: selectors.CR.Workspaces.makeIsDocumentNodeDirtySelector($get('contextPath', node))(state),
```
Comment (skurfuerst) [inline packages/neos-ui/src/Containers/LeftSideBar/NodeTree/Node/index.js]: oups, that was too much ;-) the selector *factory* must be in the outer function (at the old location); but it is *not allowed to take any arguments*.

The argument "contextPath" must be passed into the selector as 2nd parameter after "state" :-)

This way you did it here means we do not memoize at all

**Comment by dimaip** [comment]
@skurfuerst now I see what you mean by "second parameter after state". I never knew that there was such thing as a second parameter to a selector. Actually I think the convention is to pass props there, so it would work with default mapStateToProps. Maybe it would be better to do it this way everywhere -- pass the props and let the selector get what it needs?
If you don't think it's a good idea just merge it as is now.

**Comment by dfeyer** [comment]
For me it work nicely for the structure tree, but show nothing in the document tree (if I edit a document title, or for document that content unpublished edited content

File: packages/neos-ui-redux-store/src/CR/Workspaces/selectors.js [hunk 3]
```diff
@@ -14,3 +14,21 @@ export const publishableNodesInDocumentSelector = createSelector(
     ],
     (publishableNodes, activeDocumentContextPath) => publishableNodes.filter(i => $get('documentContextPath', i) === activeDocumentContextPath)
 );
+
+export const makeIsDocumentNodeDirtySelector = documentContextPath => createSelector(
```
Comment (skurfuerst) [inline packages/neos-ui-redux-store/src/CR/Workspaces/selectors.js]: this selector still needs to be rewritten (to same way as the selector below  regarding arguments :) )

**Comment by skurfuerst** [comment]
@dfeyer absolutely, can be explained by the code position I highlighted :)

**Comment by dimaip** [comment]
@skurfuerst argh sorry, I got so caught up in the conceptual reselect topics that I forgot about the second selector! Fixed.

**Pull request merged**